### PR TITLE
DRAFT #1213 added example of rowbuilder interface

### PR DESCRIPTION
**Pull request by chrisjstevo**

### CODE REVIEW

**Comment by rumakt** [comment]
Looking good.

File: vuu/src/main/scala/org/finos/vuu/core/row/InMemMapRowBuilder.scala [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+    this
+  }
+
+  override def setBoolean(column: Column, v: Boolean): RowBuilder = {
+    mutableMap.put(column.name, v)
+    this
+  }
+  override def setKey(key: String): RowBuilder = {
+    this.key = key
+    this
```
Comment (junaidzm13) [inline vuu/src/main/scala/org/finos/vuu/core/row/InMemMapRowBuilder.scala:36]: Not sure if it applies here but would be worthwhile to think about:

It uses the same instance of the row builder so if somewhere we instantiate a row builder and share the same builder with other functions there is a chance that we might mutate the data in multiple places resulting in unexpected data changes and hard to find bugs.


For example:
```
def main() = {
  ...
  val rowBuilder = InMemMapRowBuilder().setKey("key")
  function1(rowBuilder)
  function2(rowBuilder)
}

def function1(rowBuilder: RowBuilder, ...) = {
  ...
  rowBuilder.setLong(col1, 50).asRow()
  ...
}

def function2(rowBuilder: RowBuilder) = {
  ...
  // key will be "" here, and col1 will have a value of 50
  rowBuilder.setString(col2, "ABC").asRow()
  ...
}
```

File: vuu/src/main/scala/org/finos/vuu/core/row/InMemMapRowBuilder.scala [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+import org.finos.vuu.core.table.{Column, RowData, RowWithData}
+
+import scala.collection.mutable
+
+class InMemMapRowBuilder extends RowBuilder {
+
+  private val mutableMap = new mutable.HashMap[String, Any]()
+  private var key: String = ""
+  override def setLong(column: Column, v: Long): RowBuilder = {
+    mutableMap.put(column.name, v)
```
Comment (junaidzm13) [inline vuu/src/main/scala/org/finos/vuu/core/row/InMemMapRowBuilder.scala:11]: For the future: might be able to check for type consistency by matching the type i.e. `Long` with `column.dataType`.

**Review by junaidzm13 — COMMENTED** [review]
Looks good! Left a couple of comments, let me know what you think

File: vuu/src/main/scala/org/finos/vuu/core/row/InMemMapRowBuilder.scala (hunk 1 above)
Comment (chrisjstevo) [reply vuu/src/main/scala/org/finos/vuu/core/row/InMemMapRowBuilder.scala:36]: Yes, that is very true. Each caller should getetter own I think, but I think it's fair to assume different call sites shouldn't pass them across threads.
Will update and add notthreadsafe annotation

File: vuu/src/main/scala/org/finos/vuu/core/row/InMemMapRowBuilder.scala (hunk 2 above)
Comment (chrisjstevo) [reply vuu/src/main/scala/org/finos/vuu/core/row/InMemMapRowBuilder.scala:11]: Yes, that was indeed my goal. So checking datatype of column matches call signature.
I will add that.

**Pull request merged**